### ISSUE TITLE
conda install --use-local doesn't find/use my local package build 

### ISSUE THREAD

**Issue by monocongo** [conda/conda#7758]
## Current Behavior

I've built a conda package from the PyPI package using `conda-build`, which appears to have worked as expected, i.e. I can see the *.bz2 file under the environment's local builds directory for my platform. Next, I try to install the package into the environment using `conda install --use-local` and here it fails with messages indicating that it's looked for the package externally and not found it.

### Steps to Reproduce

```
$ conda skeleton pypi climate_indices

$ ll climate_indices/
total 6
-rw-r--r-- 1 DELL 197121  107 Sep 10 10:02 bld.bat
-rw-r--r-- 1 DELL 197121   81 Sep 10 10:02 build1.sh
-rw-r--r-- 1 DELL 197121 1088 Sep  9 15:29 meta.yaml

$ conda-build climate_indices

$ ls -l /c/home/miniconda3/envs/packaging/conda-bld/win-64/
total 73
-rw-r--r-- 1 DELL 197121 58736 Sep 10 13:11 climate_indices-1.0.0-py37h39e3cac_0.tar.bz2
-rw-r--r-- 1 DELL 197121  2091 Sep 10 13:11 index.html
-rw-r--r-- 1 DELL 197121   707 Sep 10 13:11 repodata.json
-rw-r--r-- 1 DELL 197121   417 Sep 10 13:11 repodata.json.bz2
-rw-r--r-- 1 DELL 197121   825 Sep 10 13:11 repodata2.json

$ conda install --use-local climate_indices
Solving environment: ...working... failed

PackagesNotFoundError: The following packages are not available from current channels:

  - climate_indices

Current channels:

  - https://repo.anaconda.com/pkgs/main/win-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/free/win-64
  - https://repo.anaconda.com/pkgs/free/noarch
  - https://repo.anaconda.com/pkgs/r/win-64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/pro/win-64
  - https://repo.anaconda.com/pkgs/pro/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-64
  - https://repo.anaconda.com/pkgs/msys2/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

## Expected Behavior

The package should install into the current environment.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : packaging
    active env location : C:\home\miniconda3\envs\packaging
            shell level : 1
       user config file : C:\Users\DELL\.condarc
 populated config files :
          conda version : 4.5.11
    conda-build version : not installed
         python version : 3.6.5.final.0
       base environment : C:\home\miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/win-64
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\home\miniconda3\pkgs
                          C:\Users\DELL\AppData\Local\conda\conda\pkgs
       envs directories : C:\home\miniconda3\envs
                          C:\Users\DELL\AppData\Local\conda\conda\envs
                          C:\Users\DELL\.conda\envs
               platform : win-64
             user-agent : conda/4.5.11 requests/2.19.1 CPython/3.6.5 Windows/10 Windows/10.0.17134
          administrator : False
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
No ouput -- is this normal?
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at C:\home\miniconda3\envs\packaging:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                   py37_0    defaults
beautifulsoup4            4.6.3                    py37_0    defaults
blas                      1.0                         mkl    defaults
bzip2                     1.0.6                    vc14_1  [vc14]  conda-forge
ca-certificates           2018.8.24            ha4d7672_0    conda-forge
certifi                   2018.4.16                py37_0    conda-forge
cffi                      1.11.5           py37h74b6da3_1    defaults
cftime                    1.0.0b1          py37h452e1ab_0    defaults
chardet                   3.0.4                    py37_1    defaults
conda                     4.5.11                   py37_0    defaults
conda-build               3.14.2                   py37_0    defaults
conda-env                 2.6.0                         1    defaults
cryptography              2.3.1            py37h74b6da3_0    defaults
curl                      7.60.0                   vc14_0  [vc14]  conda-forge
cython                    0.28.5           py37h6538335_0    defaults
filelock                  3.0.6                    py37_0    defaults
glob2                     0.6                      py37_0    defaults
hdf4                      4.2.13                   vc14_0  [vc14]  conda-forge
hdf5                      1.10.2                   vc14_0  [vc14]  conda-forge
icc_rt                    2017.0.4             h97af966_0    defaults
idna                      2.7                      py37_0    defaults
intel-openmp              2018.0.3                      0    defaults
jinja2                    2.10                     py37_0    defaults
jpeg                      9b                       vc14_2  [vc14]  conda-forge
krb5                      1.14.6                   vc14_0  [vc14]  conda-forge
libcurl                   7.61.0               h7602738_0    defaults
libnetcdf                 4.6.1                h62daf8c_0    defaults
libssh2                   1.8.0                hd619d38_4    defaults
llvmlite                  0.24.0           py37h6538335_0    defaults
markupsafe                1.0              py37hfa6e2cd_1    defaults
menuinst                  1.4.14           py37hfa6e2cd_0    defaults
mkl                       2018.0.3                      1    defaults
mkl_fft                   1.0.4            py37h1e22a9b_1    defaults
mkl_random                1.0.1            py37h77b88f5_1    defaults
netcdf4                   1.4.1            py37hbfe741f_0    defaults
numba                     0.39.0           py37h830ac7b_0    defaults
numpy                     1.15.1           py37ha559c80_0    defaults
numpy-base                1.15.1           py37h8128ebf_0    defaults
openssl                   1.0.2p               hfa6e2cd_0    conda-forge
pandas                    0.23.4           py37h830ac7b_0    defaults
pip                       18.0                      <pip>
pip                       10.0.1                   py37_0    defaults
pkginfo                   1.4.2                    py37_1    defaults
psutil                    5.4.7            py37hfa6e2cd_0    defaults
pycosat                   0.6.3            py37hfa6e2cd_0    defaults
pycparser                 2.18                     py37_1    defaults
pyopenssl                 18.0.0                   py37_0    defaults
pysocks                   1.6.8                    py37_0    defaults
python                    3.7.0                hea74fb7_0    defaults
python-dateutil           2.7.3                    py37_0    defaults
python-wget               3.2                        py_0    conda-forge
pytz                      2018.5                   py37_0    defaults
pywin32                   223              py37hfa6e2cd_1    defaults
pyyaml                    3.13             py37hfa6e2cd_0    defaults
requests                  2.19.1                   py37_0    defaults
requests-toolbelt         0.8.0                    py37_0    defaults
ruamel_yaml               0.15.46          py37hfa6e2cd_0    defaults
scipy                     0.1                       <pip>
setuptools                40.2.0                   py37_0    defaults
six                       1.11.0                   py37_1    defaults
tqdm                      4.25.0           py37h28b3542_0    defaults
twine                     1.11.0                   py37_0    defaults
urllib3                   1.23                     py37_0    defaults
vc                        14                   h0510ff6_3    defaults
vs2015_runtime            14.0.25123                    3    defaults
wheel                     0.31.1                   py37_0    defaults
win_inet_pton             1.0.1                    py37_1    defaults
wincertstore              0.2                      py37_0    defaults
yaml                      0.1.7                    vc14_0  [vc14]  conda-forge
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge
```
</p></details>

**Comment by msarahan** [comment]
Try ``-c local`` instead of ``--use-local``.  These should be equivalent, but I'm not certain that they are.

**Comment by kalefranz** [comment]
I still don't quite understand what's going on.  Maybe there's a bug in `--use-local`.

**Comment by kalefranz** [comment]
Could you share the output of `conda info --all`?

**Comment by monocongo** [comment]
Thanks, Kale.

```
$ conda info --all

     active environment : packaging
    active env location : C:\home\miniconda3\envs\packaging
            shell level : 1
       user config file : C:\Users\DELL\.condarc
 populated config files :
          conda version : 4.5.11
    conda-build version : not installed
         python version : 3.6.5.final.0
       base environment : C:\home\miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/win-64
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\home\miniconda3\pkgs
                          C:\Users\DELL\AppData\Local\conda\conda\pkgs
       envs directories : C:\home\miniconda3\envs
                          C:\Users\DELL\AppData\Local\conda\conda\envs
                          C:\Users\DELL\.conda\envs
               platform : win-64
             user-agent : conda/4.5.11 requests/2.19.1 CPython/3.6.5 Windows/10 Windows/10.0.17134
          administrator : False
             netrc file : None
           offline mode : False

# conda environments:
#
base                     C:\home\miniconda3
packaging             *  C:\home\miniconda3\envs\packaging
pkgtest                  C:\home\miniconda3\envs\pkgtest

sys.version: 3.6.5 |Anaconda, Inc.| (default, Mar 29 ...
sys.prefix: C:\home\miniconda3
sys.executable: C:\home\miniconda3\python.exe
conda location: C:\home\miniconda3\lib\site-packages\conda
conda-build: C:\home\miniconda3\envs\packaging\Scripts\conda-build.exe
conda-env: C:\home\miniconda3\Scripts\conda-env.exe
user site dirs: C:\Users\DELL\AppData\Roaming\Python\Python36

ACLOCAL_PATH: C:\Git\mingw64\share\aclocal;C:\Git\usr\share\aclocal
CIO_TEST: <not set>
CONDA_DEFAULT_ENV: packaging
CONDA_EXE: C:/home/miniconda3/Scripts/conda.exe
CONDA_PREFIX: C:\home\miniconda3\envs\packaging
CONDA_PROMPT_MODIFIER: (packaging)
CONDA_PYTHON_EXE: C:/home/miniconda3/python.exe
CONDA_ROOT: C:\home\miniconda3
CONDA_SHLVL: 1
EXEPATH: C:\Git
HOMEPATH: \Users\DELL
INFOPATH: C:\Git\usr\local\info;C:\Git\usr\share\info;C:\Git\usr\info;C:\Git\share\info
MANPATH: C:\Git\mingw64\share\man;C:\Git\usr\local\man;C:\Git\usr\share\man;C:\Git\usr\man;C:\Git\share\man
PATH: C:\home\miniconda3\envs\packaging;C:\home\miniconda3\envs\packaging\Library\mingw-w64\bin;C:\home\miniconda3\envs\packaging\Library\usr\bin;C:\home\miniconda3\envs\packaging\Library\bin;C:\home\miniconda3\envs\packaging\Scripts;C:\home\miniconda3\envs\packaging\bin;C:\home\miniconda3;C:\home\miniconda3\scripts;C:\home\miniconda3\Library\bin;C:\Users\DELL\bin;C:\Git\mingw64\bin;C:\Git\usr\local\bin;C:\Git\usr\bin;C:\Git\usr\bin;C:\Git\mingw64\bin;C:\Git\usr\bin;C:\Users\DELL\bin;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\ProgramData\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0;C:\Program Files (x86)\Intel\OpenCL SDK\2.0\bin\x86;C:\Program Files (x86)\Intel\OpenCL SDK\2.0\bin\x64;C:\scala\bin;C:\WINDOWS\System32\OpenSSH;C:\Program Files (x86)\NetCDF;C:\Apache\apache-maven-3.3.9\bin;C:\Users\DELL\.babun;C:\Users\DELL\AppData\Local\Microsoft\WindowsApps;C:\Git\usr\bin\vendor_perl;C:\Git\usr\bin\core_perl
PKG_CONFIG_PATH: C:\Git\mingw64\lib\pkgconfig;C:\Git\mingw64\share\pkgconfig
PSMODULEPATH: C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\
REQUESTS_CA_BUNDLE: <not set>
SSL_CERT_FILE: <not set>


WARNING: could not import _license.show_info
# try:
# $ conda install -n root _license
```

**Comment by monocongo** [comment]
Here's what happened when I tried the install from file as suggested above:

```
$ conda install -c file:///c/home/miniconda3/envs/packaging/conda-bld climate_indices
Solving environment: ...working... failed

CondaHTTPError: HTTP 404 CONNECTION FAILED for url <file:///c/home/miniconda3/envs/packaging/conda-bld/noarch/repodata.json>
Elapsed: 00:00.000122

The remote server could not find the noarch directory for the
requested channel with url: file:///c/home/miniconda3/envs/packaging/conda-bld

As of conda 4.3, a valid channel must contain a `noarch/repodata.json` and
associated `noarch/repodata.json.bz2` file, even if `noarch/repodata.json` is
empty. please request that the channel administrator create
`noarch/repodata.json` and associated `noarch/repodata.json.bz2` files.
$ mkdir noarch
$ echo '{}' > noarch/repodata.json
$ bzip2 -k noarch/repodata.json

You will need to adjust your conda configuration to proceed.
Use `conda config --show channels` to view your configuration's current state.
Further configuration help can be found at <https://conda.io/docs/config.html>.
```

Also, in case this is helpful:
```
$ ll /c/home/miniconda3/envs/packaging/conda-bld
total 17
-rw-r--r-- 1 DELL 197121  932 Sep 10 13:11 channeldata.json
drwxr-xr-x 1 DELL 197121    0 Sep  9 15:28 icons/
-rw-r--r-- 1 DELL 197121 2311 Sep 10 16:42 index.html
drwxr-xr-x 1 DELL 197121    0 Sep  9 15:28 noarch/
-rw-r--r-- 1 DELL 197121  600 Sep 10 16:42 rss.xml
drwxr-xr-x 1 DELL 197121    0 Sep  9 15:26 skeleton_1536521199167/
drwxr-xr-x 1 DELL 197121    0 Sep 10 16:25 skeleton_1536611136412/
drwxr-xr-x 1 DELL 197121    0 Sep 10 13:11 src_cache/
drwxr-xr-x 1 DELL 197121    0 Sep 10 16:42 win-64/
drwxr-xr-x 1 DELL 197121    0 Sep  9 15:26 work/

$ ll /c/home/miniconda3/envs/packaging/conda-bld/win-64/
total 73
-rw-r--r-- 1 DELL 197121 58596 Sep 10 16:42 climate_indices-1.0.0-py37h39e3cac_0.tar.bz2
-rw-r--r-- 1 DELL 197121  2091 Sep 10 16:42 index.html
-rw-r--r-- 1 DELL 197121   707 Sep 10 16:42 repodata.json
-rw-r--r-- 1 DELL 197121   418 Sep 10 16:42 repodata.json.bz2
-rw-r--r-- 1 DELL 197121   825 Sep 10 16:42 repodata2.json

$ ll /c/home/miniconda3/envs/packaging/conda-bld/noarch/
total 7
-rw-r--r-- 1 DELL 197121 1782 Sep  9 15:28 index.html
-rw-r--r-- 1 DELL 197121  103 Sep  9 15:28 repodata.json
-rw-r--r-- 1 DELL 197121  117 Sep  9 15:28 repodata.json.bz2
-rw-r--r-- 1 DELL 197121  120 Sep  9 15:28 repodata2.json
```

**Comment by kalefranz** [comment]
Guess per https://blogs.msdn.microsoft.com/ie/2006/12/06/file-uris-in-windows/, might need to be this?

   conda install -c file:///c:/home/miniconda3/envs/packaging/conda-bld climate_indice

**Comment by bt-** [comment]
@monocongo, did you find a solution for this issue?  I am having the same issue, but on a different platform - ox64.

Similar to your experience, I am able to successfully build the conda package and it looks like it does successfully test importing my package during the build process, but I cannot find a way to then test the local stored conda package by installing in a new environment.

I am trying to build the package ([pvcaptest](https://pypi.org/project/captest/)) locally and then test it by installing in a new environment as I envision users of the package would install.

I've tried all of the following without success:
```
(pvcaptest) [~]$$conda install --use-local pvcaptest
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - pvcaptest

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

And
```
(pvcaptest) [~]$$conda install -c local pvcaptest
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - pvcaptest

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

And
```
(pvcaptest) [~]$$conda install -c file://Users/Ben/opt/miniconda3/envs/conda_build/conda-bld/ pvcaptest
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: \ 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                                                                     

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions
```

When I use the `-vv` flag with the second option I get the following output.  This is only the top few lines.
```
(pvcaptest) [~]$$conda install -vv -c file://Users/Ben/opt/miniconda3/envs/conda_build/conda-bld/ pvcaptest
DEBUG conda.gateways.logging:set_verbosity(231): verbosity set to 2
DEBUG conda.core.solve:solve_final_state(223): solving prefix /Users/Ben/opt/miniconda3/envs/pvcaptest
  specs_to_remove: frozenset()
  specs_to_add: frozenset({MatchSpec("pvcaptest")})
  prune: <auxlib._Null object at 0x7fadae4cc290>
Collecting package metadata (current_repodata.json): ...working... DEBUG conda.core.package_cache_data:_check_writable(259):
 package cache directory '/Users/Ben/opt/miniconda3/pkgs' 
writable: True
DEBUG conda.core.subdir_data:_load(241): Local cache timed out for file://users/Ben/opt/miniconda3/envs/conda_build
/conda-bld/osx-64/current_repodata.json at /Users/Ben/opt/miniconda3/pkgs/cache/7369fe66.json
DEBUG conda.core.subdir_data:_load(241): Local cache timed out for https://repo.anaconda.com/pkgs/r/osx-64
/current_repodata.json at /Users/Ben/opt/miniconda3/pkgs/cache/3eef14e5.json
DEBUG conda.core.subdir_data:_load(241): Local cache timed out for https://repo.anaconda.com/pkgs/r/noarch
/current_repodata.json at /Users/Ben/opt/miniconda3/pkgs/cache/227b7582.json
DEBUG conda.core.subdir_data:_load(235): Using cached repodata for https://conda.anaconda.org/conda-forge/noarch
/current_repodata.json at /Users/Ben/opt/miniconda3/pkgs/cache/62c98390.json. Timeout in 102 sec
DEBUG conda.core.subdir_data:_load(241): Local cache timed out for file://users/Ben/opt/miniconda3/envs/conda_build
/conda-bld/noarch/current_repodata.json at /Users/Ben/opt/miniconda3/pkgs/cache/a8a74229.json
DEBUG conda.core.subdir_data:_load(241): Local cache timed out for https://repo.anaconda.com/pkgs/main/osx-64
/current_repodata.json at /Users/Ben/opt/miniconda3/pkgs/cache/53d7ef51.json
```

I don't actually need this package to be installed locally as I am planning for it to be hosted on the conda-forge channel, but I wanted to build and test it locally first to be sure it was doing what I expected and I had the correct minimal set of dependencies.  I will probably just try it on conda-forge next.  Is there any reason I should be building locally first?

Here is my `conda info` output:
```
     active environment : pvcaptest
    active env location : /Users/Ben/opt/miniconda3/envs/pvcaptest
            shell level : 6
       user config file : /Users/Ben/.condarc
 populated config files : /Users/Ben/.condarc
          conda version : 4.8.3
    conda-build version : not installed
         python version : 3.7.6.final.0
       virtual packages : __osx=10.15.5
       base environment : /Users/Ben/opt/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/Ben/opt/miniconda3/pkgs
                          /Users/Ben/.conda/pkgs
       envs directories : /Users/Ben/opt/miniconda3/envs
                          /Users/Ben/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.7.6 Darwin/19.5.0 OSX/10.15.5
                UID:GID : 501:20
             netrc file : /Users/Ben/.netrc
           offline mode : False
```

Here is the state of my .condarc file:
```
(pvcaptest) [~]$$cat .condarc
channels:
  - conda-forge
  - defaults
  - local
unsatisfiable_hints: true
```

**Comment by Todd-Fulton** [comment]
So when building a package in an environment, I get these errors, but, when I build from the base environment, I don't.

To be clear, lets say I'm in an environment `TESTENV` the local channel is treated as being
`file:///c:/tools/miniconda3/envs/TESTENV/conda-bld`
when placing the package on the file system.

When building the package in base after deactivating any env, the local channel is treated as being
`C:\tools\miniconda3\conda-bld`
when placing the package on the file system.

In both cases, when installing the package with `-c local` , local is expected to be in
`C:\tools\miniconda3\conda-bld`

using bare `.condarc` and after building outside of any environment (so the package is placed in `C:\tools\miniconda3\conda-bld`)
```
❯ conda install -n TESTENV-c local slixmpp
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: C:\tools\miniconda3\envs\TESTENV

  added / updated specs:
    - slixmpp


The following NEW packages will be INSTALLED:

  aiodns             pkgs/main/noarch::aiodns-2.0.0-py_2
  aiohttp            pkgs/main/win-64::aiohttp-3.6.1-py38he774522_0
  async-timeout      pkgs/main/win-64::async-timeout-3.0.1-py38_0
  multidict          pkgs/main/win-64::multidict-4.7.3-py38he774522_0
  pycares            pkgs/main/win-64::pycares-3.1.1-py38he774522_0
  slixmpp            C:/tools/miniconda3/conda-bld/win-64::slixmpp-1.5.2-py38_0
  yarl               pkgs/main/win-64::yarl-1.4.2-py38he774522_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

Given the above, I'm guessing the problem is that when inside of an environment, conda expects local to be in the usual place instead of in a local channel location that is local to the environment, I hope that makes sense.

Idk if it's fine the way it is, if so, maybe the documentation needs to make it clear this is the way it behaves and to always build from base and use `--python VERSION` to target a version different from the base install.

**Comment by angloyna** [comment]
> -rw-r--r-- 1 DELL 197121 58596 Sep 10 16:42 climate_indices-1.0.0-py37h39e3cac_0.tar.bz2
> python version : 3.6.5.final.0

I wonder if @monocongo 's issue were as simple as a python version compatibility mismatch

**Comment by monocongo** [comment]
I apologize I haven't been able to focus on this for almost a couple of years now. Perhaps now is a good time for me to revisit this issue, hopefully I'll get some bandwidth soon...

**Comment by krinsman** [comment]
This isn't much of a workaround (I have the same problem and don't think I've been able to fix it), but what I do is the following:

awhile ago I followed the instructions here

https://docs.conda.io/projects/conda-build/en/latest/user-guide/tutorials/build-pkgs.html#optional-uploading-new-packages-to-anaconda-org

such that whenever I build a new package, it gets uploaded to my personal anaconda channel

So then I can do `conda install -c krinsman <package_name>`.

Admittedly though this isn't a great workaround since it requires creating an Anaconda account, means that your built packages will always get uploaded to Anaconda regardless of whether you intended to, etc. etc. Also won't work if you don't have an internet connection although I suppose conda in general doesn't work without an internet connection so

Sorry for the dumb suggestion and not being more helpful -- I would like a better solution as well
